src/include/daos_task.h

```diff
@@ -326,6 +326,8 @@ typedef struct {
 	daos_prop_t		*prop;
 	/** Optional returned the allocated container UUID */
 	uuid_t			*cuuid;
+	/** Time of the request (hybrid logical clock). */
```

Comment:
@mchaarawi pointed out this may not be where we want to be keeping the request timestamps, since changing these structures changes the API. I will take a look at embedding the timestamp in alternate structures "behind the scenes" within the implementation, if that is possible.

The requirement is that the timestamp be assigned on the initial RPC request, and preserved for any new RPCs sent if the operation is retried.

Reply:
every task has a private struct associated with it. I wonder if you could extend that to include this, which in that case would not be exposed to the user.
this request timestamp seems generic enough that it could apply to all task operations, even though it will only be used for this container operations today. what do you think?

Reply:
If you mean `dc_task_set_priv`/`daos_task_set_priv` then I agree. The pattern looks like:

```
dc_some_operation(task):
  if task private pointer is null
    check input parameters
    allocate operation key
    ...
    initialize the private pointer
  choose PS/CS rank
  send RPC
```